nvflare/fuel/f3/drivers/uds_driver.py

```diff
@@ -65,14 +65,12 @@ def listen(self, connector: ConnectorInfo):
         self.connector = connector
 
         socket_path = self.get_socket_path(connector.params)
-        try:
-            os.unlink(socket_path)
-        except OSError:
-            if os.path.exists(socket_path):
-                raise CommError(CommError.ERROR, f"Can't remove existing socket file: {socket_path}")
+        if not self.remove_socket_file(socket_path):
+            raise CommError(CommError.ERROR, f"Can't remove existing socket file: {socket_path}")
 
         self.server = SocketStreamServer(socket_path, self, connector)
         self.server.serve_forever()
+        self.remove_socket_file(socket_path)
```

Comment:
Will both the "connector" side and "listener" side try to remove the same socket file at the same time?

Reply:
It may happen. Only the last one who closes the connection can remove the file, which can be either side. So both sides try to remove the file. Only one will succeed.